### DISCUSSION TITLE
✅ Answered/Solved: Joomla 3+4- Feature Request: Improve Actionlog search

### DISCUSSION THREAD

**Discussion by coolcat-creations** [joomla/joomla-cms#36013] [2. Cold Features]
### Steps to reproduce the issue
Have Actionlog activated
Change an article or category or whatever
Use the Search Box in the user Action log to find who changed article or category xyz

### Expected result
Search should show a list of actions that have been done to the object you search

### Actual result
Search does not find anything altough its there

### System information (as much as possible)
Happens in Joomla 4 and in 3

**Comment by brianteeman** [comment]
What exactly is your search term?

> Search in username. Prefix with ID: to search for an actionlog ID. Prefix with ITEM_ID: to search for an actionlog item ID.

I just did a quick test of the above in J3 and it seemed to work ok

**Comment by coolcat-creations** [comment]
Example:
![grafik](https://user-images.githubusercontent.com/828371/141307474-bd35b18f-9d9f-4ab4-aa25-0590131e3c7b.png)


![grafik](https://user-images.githubusercontent.com/828371/141307413-310511ce-63ef-40ea-b971-a1af8c63d676.png)

**Comment by coolcat-creations** [comment]
Second example

![grafik](https://user-images.githubusercontent.com/828371/141307611-69e13089-9f9d-4892-aeb2-9f43e34dfe4e.png)
![grafik](https://user-images.githubusercontent.com/828371/141307702-c0d6e295-d531-43fc-b28e-c4b0a337c712.png)

**Comment by brianteeman** [comment]
Rightly or wrongly that would be the expected behaviour according to the information.

> Search in username. Prefix with ID: to search for an actionlog ID. Prefix with ITEM_ID: to search for an actionlog item ID.

**Comment by ChristineWk** [comment]
confirm last images.
hv tried another action. maybe compromise

![screen shot 2021-11-11 at 14 42 20](https://issues.joomla.org/uploads/1/96e911679982b441d9393e44c7124902.png)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36009">issues.joomla.org/tracker/joomla-cms/36009</a>.</sub>

**Comment by richard67** [comment]
> Rightly or wrongly that would be the expected behaviour according to the information.
> 
> > Search in username. Prefix with ID: to search for an actionlog ID. Prefix with ITEM_ID: to search for an actionlog item ID.

Then I would say this issue should be closed as expected behavior or be changed to a feature request.

**Comment by alikon** [comment]
moved to discussions

not an issue but a new feature... imho

**Comment by ChristineWk** [comment]
the problem is, you would first have to select from the list plus enter the desired user and that's laborious. 

![Screenshot 2021-11-11 223314](https://user-images.githubusercontent.com/15275917/141373665-0e708e7b-d561-4aff-8920-6500cc5fad89.png)

so it probably needs a new feature or change. but of course I don't know how.

**Comment by coolcat-creations** [comment]
Thanks for moving to discussions instead of closing. I actually expected the search to find titles that are displayed in the search - did not consider that its not a bug. How can the search be extended to the items?

**Reply by brianteeman** [reply]
Please review #36242 I think this is what you are looking for

**Reply by brianteeman** [reply]
as #36242 has now been merged please can you mark this discussion as answered

**Reply by coolcat-creations** [reply]
Thank you, done!
